sw/device/tests/sim_dv/pwrmgr_deep_sleep_all_wake_ups.c

```diff
@@ -0,0 +1,294 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_adc_ctrl.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/dif/dif_sysrst_ctrl.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+/*
+  PWRMGR DEEP SLEEP ALL WAKE UPS test
+
+  This test runs power manager wake up from deep leep mode by
+  wake up inputs.
+
+  There are 6 wake up inputs.
+  0: sysrst_ctrl
+  1: adc_ctrl
+  2: pinmux
+  3: usb
+  4: aon_timer
+  5: sensor_ctrl
+
+  #3 and 4 are covered by separate test (see
+  https://github.com/lowRISC/opentitan/blob/master/
+  hw/top_earlgrey/data/chip_testplan.hjson)
+
+  #5 is excluded because sensor_ctrl is not aon domain
+
+ */
+#define kTestNumWkup 3
+#define kNoWkupDev 0
+#define kPinmuxWkupDetector5 5
+
+#define MAKE_INIT_FUNC(dif_)                                             \
+  void dif_##_init(void *dif, uintptr_t base) {                          \
+    CHECK_DIF_OK(dif##_##dif_##_init(mmio_region_from_addr(base), dif)); \
```

Comment:
Can we just call the DIF directly, instead of wrap it in a function that is then wrapped by a macro?

Reply:
this wrapper needs for // device init  loop in test_main function.
without wrapper, loops doesn't work and had to write each init
individually.

Reply:
I think I would drop the macro wrapper for the init functions, between the loop and this macro, its about the same amount of LOC anyway, and it will increase the readability.

Reply:
fixed